swift/ql/src/queries/Security/CWE-089/SqlInjection.qhelp

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+
+<p>
+If a database query (such as an SQL query) is built from user-provided data without sufficient sanitization, a user may be able to run malicious database queries.
+</p>
+
+</overview>
+<recommendation>
+
+<p>
+Most database connector libraries offer a way of safely embedding untrusted data into a query by means of query parameters or prepared statements. Use these features rather than building queries by string concatenation.
+</p>
+
+</recommendation>
+<example>
+
+<p>In the following example, an SQL query is prepared using string interpolation to directly include a user-controlled value <code>userControlledString</code> in the query. An attacker could craft the part they control to change the overall meaning of the SQL query.
+</p>
+
+<sample src="SqlInjectionBad.swift" />
+
+<p>A better way to do this is with a prepared statement, binding <code>userControlledString</code> to a parameter of that statement. An attacker who controls the contents of that parameter cannot 'break out' and change the overall meaning of the SQL query.
```

Comment:
Is there a different way of expressing the "break out" part? Unless this is a technical term related to SQL then it'd be preferable to avoid metaphors and expressions (basically anything in inverted commas) so as to help comprehension, particularly for people for whom English is a second language 🙏

Reply:
Yeah, I can see that this might be confusing for someone who doesn't speak English as a first language.  I've removed the metaphor.